java/src/main/java/com/factset/protobuf/stach/extensions/Utilities.java

```diff
@@ -12,4 +19,84 @@ public static boolean isNullOrEmpty(String str) {
         return str == null || str.isEmpty();
     }
 
+    /**
+     * Get a list of primary table ids from stach data
+     *
+     * @param pkg : The stach data.
+     * @return Returns a list of primary table ids
+    */
+    public static List<String> getPrimaryTableIds(String pkg) {
+        JSONObject pkgJson = new JSONObject(pkg);
+
+        List<Object> ids = pkgJson.getJSONArray("primaryTableIds").toList();
+
+        List<String> retList = new ArrayList<String>();
+        ids.forEach(value -> retList.add((String)value));
+
+        return retList;
+    }
+
+    /**
+     * Decompress stach data. Goes through every primary table id.
+     *
+     * @param pkg : The stach data.
+     * @return Returns a list of primary table ids
+    */
+    public static String decompress(String pkg) {
+        Iterator<String> primaryTableIds = getPrimaryTableIds(pkg).iterator();
+        JSONObject pkgJson = new JSONObject(pkg);
+
+        while(primaryTableIds.hasNext()) {
+            String tableId = primaryTableIds.next();
+            
+            decompress(pkgJson, tableId);
+        }
+
+        return pkgJson.toString();
+    }
```

Comment:
These utils methods seem to be specific to column organized STACH V2. Can you please raise another PR and move them to a separate class? The same has been pointed out in the `python` PR too and was addressed.

@Ramanth FYI

Reply:
Okay will open another PR for this.